docs/rfc/0698-cli-infrastructure.adoc

```diff
@@ -0,0 +1,295 @@
+core commands that we wish to include for the first iteration. More
+components MAY be added as the result of future RFCs. We SHALL keep a
+list of the core commands as part of the `radicle-link` documentation.
+
+=== `rad-identities`
+
+The `rad identities` subcommand provides the management of identity
+documents and their metadata. It is further split into subcommands
+which we sketch out below:
+
+* `project [ create | get | list | update | checkout | review ]` --
+  modify and retrieve `Project` identities.
+* `person [ create | get | list | update | checkout | review ]` --
+  modify and retrieve `Person` identities.
+* `any [ get | list ]` -- retrieve identities where the caller does not
+  know what kind of identity it is. The CLI will make a best-effort to
+  infer the type, or else return an opaque blob.
+* `local [ set | get | default ]` -- modify or retrieve a local identity
+* `tracking [ track | untrack | list ]` -- modify and list the tracking
+  graph for a given identity.
+* `rad-refs [ self | signed | delegates | delegate ]` -- retrieve the
+  blobs that live under `rad/<refname>`
+* `refs [ heads | notes | tags | category ]` -- list the reference names.
+  under the respective category, where `category` is a user supplied category.
+
+=== `rad-replicate`
+
+* `ask` -- tell the `daemon` that you wish to replicate the
+  given URN. Pending <<rl-141, radicle-link#141>>.
+* `from` -- if the peer/address are known for the given URN,
+  attempt to replicate the identity. This requires the activation of
+  the peer-to-peer daemon.
+
+=== `rad-profile`
+
+This command will provide the management of keys and the `Profile`
+construct in `librad`. The following actions will be defined:
+
+* `create`— generate a new `Profile`-- setting up a new key and any
+  related infrastructure for the profile, e.g. storage.
+* `get`— get the current `Profile`'s identifier.
+* `set`— set the current `Profile` to the identifier.
+  passed in. *Note* that this should refuse to do so if the daemon is online.
+* `list`— list the available `Profile` identifiers.
+* `ssh-add`— add the current `Profile`'s key to the `ssh-agent`.
+
+=== `rad-daemon`
+
+This command provides introspection into the peer-to-peer daemon.
+
+* `peers` -- get the list of currently connected peers
+* `membership` -- get the membership list for the peer-to-peer
+  daemon's membership view.
+* `stats` -- request the stats of the peer-to-peer daemon, i.e. a
+  summary of peers and members.
+
+=== Note on Output
+
+Some operations will result in data that should be output, for
+example, retrieving a `Project` from the store will return the
+metadata for that project. As per <<arch-cli, RFC-682#CLI>>:
+
+[quote]
+-------
+
+Each subcommand MUST expose its functionality as a linkable library,
+and provide CBOR serialisation for its arguments and outputs.
+-------
+
+The output should, however, not be limited to CBOR. We will provide an
```

Comment:
Note to self: CBOR actually is mentioned, need to add the link back https://datatracker.ietf.org/doc/html/rfc8949